src/Microsoft.TestPlatform.ManagedNames/Extensions/TestCase2Extensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Extensions
+{
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    public static class TestCase2Extensions
+    {
+        /// <summary>
+        /// Returns <c>true</c> if both <see cref="ITestCase2.ManagedType" /> and <see cref="ITestCase2.ManagedMethod" /> are not null or whitespace.
+        /// </summary>
+        public static bool HasManagedMethodAndType(this ITestCase2 testCase)
```

Comment:
Assuming we cannot change the `ITestCase2` interface, maybe we should consider introducing `ITestCase3` that inherits `ITestCase2` and adds the new methods/properties instead of working around with extension methods ?

Reply:
`ITestCase2` interface is new. `ContainsManagedMethodAndType` is not included there; I'll include it. The interface is explicitly implemented, so these extension methods provide ease of access to those members.